src/Dumplie/Domain/Customer/Cart.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Domain\Customer;
+
+use Dumplie\Domain\Customer\Exception\ProductNotAvailableException;
+use Dumplie\Domain\SharedKernel\Exception\InvalidCurrencyException;
+use Dumplie\Domain\SharedKernel\Money\Price;
+
+final class Cart
+{
+    /**
+     * @var CartId
+     */
+    private $id;
+
+    /**
+     * @var CartItem[]|array
+     */
+    private $items;
+
+    public function __construct()
+    {
+        $this->id = CartId::generate();
+        $this->items = [];
+    }
+
+    /**
+     * @return CartId
+     */
+    public function id() : CartId
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param Product $product
+     * @param int $quantity
+     * @throws \Dumplie\Domain\SharedKernel\Exception\InvalidCurrencyException
+     * @throws \Dumplie\Domain\Customer\Exception\ProductNotAvailableException
+     */
+    public function add(Product $product, int $quantity)
+    {
+        if (!$product->isAvailable()) {
+            throw new ProductNotAvailableException($product);
+        }
+
+        foreach ($this->items as $item) {
+            if (!$item->totalPrice()->hasSameCurrency($product->price())) {
+                throw new InvalidCurrencyException($item->totalPrice()->currency(), $product->price()->currency());
+            }
+        }
+
+        $this->items[$product->sku()] = new CartItem($product, $quantity);
```

Comment:
It would require to add commands that reflect those methods, for now I would like to not complicate it too much.